packages/tools/src/stateManagement/annotation/annotationHide.ts

```diff
@@ -0,0 +1,178 @@
+import { eventTarget, triggerEvent } from '@cornerstonejs/core';
+import { Events } from '../../enums';
+import { Annotation } from '../../types';
+import { AnnotationHideChangeEventDetail } from '../../types/EventTypes';
+
+/*
+ * Constants
+ */
+const globalHiddenAnnotationsSet: Set<Annotation> = new Set();
```

Comment:
I know you did the same logic as locking, but we don't need the full `annotation` for hiding it - It will cause a lot of headache in the consumer as it needs the full annotation - `AnnotationUID` is enough in fact. I have revised the annotaiton selection, and fixing locking was on my todo list. Follow the `annotationSelection` I would recommend

Reply:
Agreed, this should use UIDs. In practice we found you always have the UID, and in this old style used to always call a get of the annotation from the UID, before passing it in to these methods.

Reply:
Ok I felts the same :D, I will change thanks